### PR TITLE
make a mock for Randomizer so I can test the function

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: TestingRandomTests/DiceRollerTests.cs [hunk 1]
```diff
@@ -23,22 +24,23 @@ public void Roll_Throws_Format_Exception_For_Malformed_Input_String(string input
...
-            for (var i = 0; i < 100; i++)
-            {
-                var result = _sut.Roll("1d6");
-                if (result < 1 || 6 < result)
-                {
-                    throw new Exception($"Result out of range: {result}");
-                }
-            }
+            var randomMock = new Mock<IRandomizer>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>())).Returns(() => dieCount);
```
Comment (jdphenix) [inline TestingRandomTests/DiceRollerTests.cs]: Not needed.

File: TestingRandomTests/DiceRollerTests.cs [hunk 2]
```diff
@@ -23,22 +24,23 @@ public void Roll_Throws_Format_Exception_For_Malformed_Input_String(string input
...
         {
-            for (var i = 0; i < 100; i++)
-            {
-                var result = _sut.Roll("1d6");
-                if (result < 1 || 6 < result)
-                {
-                    throw new Exception($"Result out of range: {result}");
-                }
-            }
+            var randomMock = new Mock<IRandomizer>();
```
Comment (jdphenix) [inline TestingRandomTests/DiceRollerTests.cs]: Consider setting this up for the entire test suite, in the constructor.

File: TestingRandomTests/DiceRollerTests.cs [hunk 3]
```diff
@@ -12,7 +13,7 @@ public DiceRollerTests()
         {
             _sut = new DiceRoller(new Randomizer());
```
Comment (jdphenix) [inline TestingRandomTests/DiceRollerTests.cs]: Set up `_sut` with a mock.

File: TestingRandomTests/DiceRollerTests.cs [hunk 4]
```diff
@@ -23,22 +24,23 @@ public void Roll_Throws_Format_Exception_For_Malformed_Input_String(string input
...
-                    throw new Exception($"Result out of range: {result}");
-                }
-            }
+            var randomMock = new Mock<IRandomizer>();
+            randomMock.Setup(r => r.Next(It.IsAny<int>())).Returns(() => dieCount);
+            var diceRoller = new DiceRoller(randomMock.Object);
+
+            var actual = diceRoller.Roll(diceExpression);
+
+            randomMock.Verify(mock => mock.Next(6), Times.Exactly(dieCount));
```
Comment (jdphenix) [inline TestingRandomTests/DiceRollerTests.cs]: Once `_sut` is set up with a mock, the entire test method can be

    [Theory]
    [InlineData("2d6", 2, 6)]
    [InlineData("3d6", 3, 6)]
    [InlineData("1d8", 1, 8)]
    public void Roll_Never_Out_Of_Range(string diceExpression, int dieCount, int sideCount)
    {
        _ = _sut.Roll(diceExpression);
        
        _randomMock.Verify(r => r.Next(sideCount), Times.Exactly(dieCount));
    }

**Review by jdphenix — CHANGES_REQUESTED** [review]
Consider setting up the test suite's `_sut` with a mock. 

For the test itself, we don't care about the actual result. We only care about how it interacts with it's dependency (the `IRandomizer` instance).

File: TestingRandomTests/DiceRollerTests.cs (hunk 2 above)
Comment (ghost) [reply TestingRandomTests/DiceRollerTests.cs]: Oke, we did not learned how to do that. So I did that at the way we learned on the bootcamp. Kai wants to have all three steps in every test

**Review by jdphenix — APPROVED** [review]
Good work, thanks!

**Pull request closed without merging**